collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_info.py

```diff
@@ -754,6 +826,57 @@ def generate_pgroups_dict(array):
     return pgroups_info
 
 
+def generate_rl_dict(array):
+    rl_info = {}
+    api_version = array._list_available_rest_versions()
+    if ACTIVE_DR_API in api_version:
+        rlinks = array.list_pod_replica_links()
+        for rlink in range(0, len(rlinks)):
+            link_name = rlinks[rlink]['local_pod_name']
+            since_epoch = rlinks[rlink]['recovery_point'] / 1000
+            recovery_datatime = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(since_epoch))
+            rl_info[link_name] = {
+                'status': rlinks[rlink]['status'],
+                'direction': rlinks[rlink]['direction'],
+                'lag': str(rlinks[rlink]['lag'] / 1000) + 's',
+                'remote_pod_name': rlinks[rlink]['remote_pod_name'],
+                'remote_names': rlinks[rlink]['remote_names'],
+                'recovery_point': recovery_datatime,
+            }
+    return rl_info
+
+
+def generate_del_pods_dict(array):
+    pods_info = {}
+    api_version = array._list_available_rest_versions()
+    if AC_REQUIRED_API_VERSION in api_version:
+        pods = array.list_pods(mediator=True, pending_only=True)
+        for pod in range(0, len(pods)):
+            acpod = pods[pod]['name']
+            pods_info[acpod] = {
+                'source': pods[pod]['source'],
+                'arrays': pods[pod]['arrays'],
+                'mediator': pods[pod]['mediator'],
+                'mediator_version': pods[pod]['mediator_version'],
+            }
+            if pods_info[acpod]['arrays'][0]['frozen_at']:
+                frozen_time = pods_info[acpod]['arrays'][0]['frozen_at'] / 1000
+                frozen_datetime = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(frozen_time))
+                pods_info[acpod]['arrays'][0]['frozen_at'] = frozen_datetime
+            if ACTIVE_DR_API in api_version:
+                pods_info[acpod]['link_source_count'] = pods[pod]['link_source_count']
+                pods_info[acpod]['link_target_count'] = pods[pod]['link_target_count']
+                pods_info[acpod]['promotion_status'] = pods[pod]['promotion_status']
+                pods_info[acpod]['requested_promotion_state'] = pods[pod]['requested_promotion_state']
+        if PREFERRED_API_VERSION in api_version:
+            pods_fp = array.list_pods(failover_preference=True, pending_only=True)
+            for pod in range(0, len(pods_fp)):
+                acpod = pods_fp[pod]['name']
+                pods_info[acpod]['failover_preference'] = pods_fp[pod]['failover_preference']
+                pods_info[acpod]['time_remainig'] = pods_fp[pod]['time_remainig']
```

Comment:
Suggest to change `pods_fp[pod]['time_remainig'] ` to `pods_fp[pod].get('time_remainig')` for backwards compatibility with purity 5.x.

Reply:
actually a typo and in the wrong place